scripts/codegen-utils.js

```diff
@@ -83,6 +83,40 @@ function fixOldArchJavaForRN72Compat(dir) {
   });
 }
 
+/**
+  * Adapts files codegend for the new architecture for paticular needs of old architecure.
+  * This usually comes down to keeping backward compat on old architecture.
+  *
+  * @param {string} dir - directory with codegened files
+  */
+function fixOldArchJavaForRN77Compat(dir) {
+  console.log(`${TAG} fixOldArchJavaForRN77Compat:  ${dir}`);
+  const files = readdirSync(dir);
+  files.forEach(file => {
+    const filePath = path.join(dir, file);
+    const fileExtension = path.extname(file);
+    if (fileExtension === '.java') {
+      const fileContent = fs.readFileSync(filePath, 'utf-8');
+      let newFileContent = fileContent.replace(
+        /extends ViewManagerWithGeneratedInterface/g,
+        '',
+      );
+      if (fileContent !== newFileContent) {
+        // Also remove redundant import
+        newFileContent = newFileContent.replace(
+          /import com.facebook.react.uimanager.ViewManagerWithGeneratedInterface;/,
+          '',
+        );
+
+        console.log(' => fixOldArchJava77 applied to:', filePath);
+        fs.writeFileSync(filePath, newFileContent, 'utf-8');
+      }
+    } else if (fs.lstatSync(filePath).isDirectory()) {
+      fixOldArchJavaForRN77Compat(filePath);
+    }
+  });
+}
```

Comment:
78 introduced `ViewManagerWithGeneratedInterface` that codegened interfaces extend.

This causes issues on RN versions <= 77. However it seems to not be required for Paper to work properly on 78, therefore I decided to remove it.